docker-app/qfieldcloud/filestorage/tests/test_range.py

```diff
@@ -0,0 +1,144 @@
+import logging
+from io import StringIO
+
+from rest_framework import status
+from rest_framework.test import APITransactionTestCase
+
+from qfieldcloud.authentication.models import AuthToken
+from qfieldcloud.core.models import (
+    Person,
+    Project,
+)
+from qfieldcloud.core.tests.mixins import QfcFilesTestCaseMixin
+from qfieldcloud.core.tests.utils import (
+    setup_subscription_plans,
+)
+from qfieldcloud.filestorage.utils import parse_range
+
+logging.disable(logging.CRITICAL)
+
+
+class QfcTestCase(QfcFilesTestCaseMixin, APITransactionTestCase):
+    def setUp(self):
+        setup_subscription_plans()
+
+        # Create a user
+        self.u1 = Person.objects.create_user(username="u1", password="abc123")
+        self.t1 = AuthToken.objects.get_or_create(user=self.u1)[0]
+        self.project_default_storage = Project.objects.create(
+            owner=self.u1,
+            name="project_default_storage",
+            file_storage="default",
+        )
+        self.project_webdav_storage = Project.objects.create(
+            owner=self.u1,
+            name="project_webdav_storage",
+            file_storage="webdav",
+        )
+
+    def test_parsing_range_function_succeeds(self):
+        start_byte, end_byte = parse_range("bytes=4-8")
+
+        self.assertEquals(start_byte, 4)
+        self.assertEquals(end_byte, 8)
```

Comment:
btw, this is more compact and IMO readable. No strong opinion, just a hint:

```suggestion
        self.assertEquals(parse_range("bytes=4-8"), (4, 8))
```

Reply:
And a few more success cases:

```
# supports start position to be 0
self.assertEquals(parse_range("bytes=0-8"), (0, 8))
self.assertEquals(parse_range("bytes=0-"), (0, None))
# supports non single digit numbers
self.assertEquals(parse_range("bytes=10-999"), (10, 999))
```